Mathlib/Topology/Algebra/InfiniteSum/Basic.lean

```diff
@@ -1139,19 +1139,22 @@ theorem cauchySeq_finset_iff_tsum_vanishing :
       rintro _ ⟨b, -, rfl⟩
       exact hts b.prop
     · exact tsum_eq_zero_of_not_summable ht ▸ mem_of_mem_nhds ho
-  obtain ⟨s, hs⟩ := vanish _ he
-  exact ⟨s, fun t hts ↦ (t.tsum_subtype f).symm ▸ hs _ hts⟩
+  · obtain ⟨s, hs⟩ := vanish _ he
+    exact ⟨s, fun t hts ↦ (t.tsum_subtype f).symm ▸ hs _ hts⟩
 
 theorem cauchySeq_finset_iff_nat_tsum_vanishing {f : ℕ → α} :
     (CauchySeq fun s : Finset ℕ ↦ ∑ n in s, f n) ↔
       ∀ e ∈ 𝓝 (0 : α), ∃ N : ℕ, ∀ t ⊆ {n | N ≤ n}, (∑' n : t, f n) ∈ e := by
-  refine cauchySeq_finset_iff_tsum_vanishing.trans ⟨fun van e he ↦ ?_, fun van e he ↦ ?_⟩ -- slow!
-  · obtain ⟨s, hs⟩ := van e he
+  refine cauchySeq_finset_iff_tsum_vanishing.trans ⟨fun vanish e he ↦ ?_, fun vanish e he ↦ ?_⟩
+  /- This is slow because CauchySeq requires SemilatticeSup (Finset ℕ) which requires
+    DecidableEq ℕ, which is the classical instance in `cauchySeq_finset_iff_tsum_vanishing`,
+    but a constructive one here. -/
```

Comment:
@sgouezel I diagnose the slowness to be due to DecidableEq instance mismatch. Do you think I should add [DecidableEq β] to [summable_iff_cauchySeq_finset](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Topology/Algebra/InfiniteSum/Basic.html#summable_iff_cauchySeq_finset) etc.?

Reply:
> It's a bit surprising that refine worked despite the instance mismatch; a Lean 4 feature?

Actually I think it's because `CauchySeq` actually only needs `Preorder` to work (for `atTop` to make sense), which is independent of the DecidableEq instance; it also explains why `refine` is slow, as Lean needs to unfold to the Preorder. We can probably get rid of some `DecidableEq` added here once we do the refactor there.